core/Db/TransactionLevel.php

```diff
@@ -54,7 +59,13 @@ public function setUncommitted()
             $this->db->query('SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED');
             $this->statusBackup = $backup;
 
-            Option::set(self::TEST_OPTION_NAME, '1'); // try setting something w/ the new transaction isolation level
+            if ($this->db->supportsUncommitted === null) {
+                // the first time we need to check if the transaction level actually works by
+                // trying to set something w/ the new transaction isolation level
+                Option::set(self::TEST_OPTION_NAME, '1');
+            }
+            $this->db->supportsUncommitted = true;
+            
```

Comment:
when it goes in the exception below, then we need to set `$this->db->supportsUncommitted = false;`